app/controllers/concerns/signature_verification.rb

```diff
@@ -100,7 +108,8 @@ def request_body
   private
 
   def verify_signature(account, signature, compare_signed_string)
-    if account.keypair.public_key.verify(OpenSSL::Digest::SHA256.new, signature, compare_signed_string)
+    if (signature_algorithm == 'rsa-sha256' && account.keypair.public_key.verify(OpenSSL::Digest::SHA256.new, signature, compare_signed_string)) ||
+       (signature_algorithm == 'hs2019' && account.keypair.public_key.verify_pss(OpenSSL::Digest::SHA512.new, signature, compare_signed_string))
```

Comment:
The structure of this code is difficult to read and kind of confusing. Maybe we could extract this into a method? Like this:

```ruby
def signature_matches_digest?(account, signature, compare_signed_string)
  case signature_algorithm
  when 'rsa-sha256'
    account.keypair.public_key.verify(OpenSSL::Digest::SHA256.new, signature, compare_signed_string))
  when 'hs2019'
    account.keypair.public_key.verify_pss(OpenSSL::Digest::SHA512.new, signature, compare_signed_string))
  end
end
```

Reply:
for now I'm going to roll back to just the `verify`, since `hs2019` doesn't necessarily translate to RSASSA-PSS and SHA512, and I couldn't manage to verify examples anyway